pkg/v1/tkg/client/validate.go

```diff
@@ -813,39 +813,73 @@ func (c *TkgClient) ValidateVsphereVipWorkloadCluster(clusterClient clusterclien
 
 // ValidateVsphereResources validates vsphere resource path specified in tkgconfig
 func (c *TkgClient) ValidateVsphereResources(vcClient vc.Client, dcPath string) error {
+	var path, resourceMoid string
+	var err error
+
 	for _, resourceType := range VsphereResourceType {
-		path, err := c.TKGConfigReaderWriter().Get(resourceType)
+		path, err = c.TKGConfigReaderWriter().Get(resourceType)
 		if err != nil {
 			continue
 		}
 
 		switch resourceType {
+		case constants.ConfigVariableVsphereDatacenter:
```

Comment:
What is the finder's behavior when multiple resources of the same name and type is present in the vSphere inventory? 
The one situation we should definitely avoid is arbitrarily selecting one of a viable fully qualified paths in that case. 
Also a good way to explain the code's behavior is to set up the unit tests to demonstrate how such ambiguities are being handled.

Reply:
Added unit tests.